### DISCUSSION TITLE
Prisma Generate Out of Memory; Running 'Prisma Generate' for dist?

### DISCUSSION THREAD

**Discussion by chrstinalin** [prisma/prisma#20602] [Q&A]
### Question

Hi!

I'm attempting to run a dist of my Nest.JS API (following [this guide](https://www.prisma.io/blog/nestjs-prisma-rest-api-7D056s1BmOL0)) on a CentOS server.

However, I'm having some issues with Prisma that I can't seem to find information on. When I run `node main &` --

```
[Nest] 13100  - 08/08/2023, 11:16:06 AM     LOG [NestFactory] Starting Nest application...
[Nest] 13100  - 08/08/2023, 11:16:06 AM   ERROR [ExceptionHandler] @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.
```

And when I attempt `npx prisma generate` --

```
var wasmInstance = new WebAssembly.Instance(wasmModule, imports);
                       ^

RangeError: WebAssembly.Instance(): Out of memory: wasm memory

```

**Is there a way I can avoid running prisma generate on the server-side and simply upload it ready-to-go in the dist folder?**

I attempted `output = "../dist"`, but what concerns me is the`import { PrismaClient } from './dist'` --

```
✔ Generated Prisma Client (5.1.1 | library) to .\dist in 658ms
You can now start using Prisma Client in your code. Reference: https://pris.ly/d/client
```

whereas, in my code, I'm using `import { PrismaClient } from '@prisma/client'`.

I'm pretty confused. Any pointers would be appreciated!

**Comment by janpio** [comment]
Yes, if you export to a custom location you need to use the custom location to import from. Everything else works as usual.

**Comment by chrstinalin** [comment]
@janpio I ended up just copying the .prisma and @prisma/client folders into the server, and that seems to have satisfied it.

However, I'm wondering if this is the wrong way to go about it: I have it up and running, but attempting a query gives me a `PrismaClientInitializationError; Error creating a database connection.`

Could this have anything to do with my copying over like this?

**Reply by janpio** [reply]
It should not, but we have `output` specifically for such cases. `@prisma/client` and `.prisma` are combined into one folder in this case, and then you can copy wherever you want.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!

**Comment by Developer-DaveRam** [comment]
hey i get this error in  server 

RangeError: WebAssembly.Instance(): Out of memory: Cannot allocate Wasm memory for new instance
    at
    
    while deploying my project in Production while trying to create npx prisma db push as well as npx prisma generate 
    this is the 1st time every using prisma in Server
